### PR TITLE
fix 942190 : Update 942190.data and rebuild rule with crs-toolchain.

### PR DESCRIPTION
**Pull request by 53cur3M3**
Update 942190.data and rebuild rule with crs-toolchain. Needed to replace :space: :^space: due to apache incompatibility. Needed to remove nested bracket for :space: and :nospace: inside character class"
[i2856 4967794a] Update 942190.data and rebuild rule with crs-toolchain. Needed to replace :space: :^space: due to apache incompatibility. Needed to remove nested bracket for :space: and :nospace: inside character class

### CODE REVIEW

**Comment by 53cur3M3** [comment]
Should resolve : https://github.com/coreruleset/coreruleset/issues/2856
On-hold until CRS-toolchain cutover

File: regex-assembly/942190.ra [hunk 1]
```diff
@@ -3,25 +3,27 @@
...
 
-[\"'`]\s*?!\s*?[\"'`\w]
-[\"'`];?\s*?having\b\s*?[^\s]
-[\"'`];?\s*?select\b\s*?[^\s]
-[\"'`];?\s*?union\b\s*?all\b\s*?[^\s]
-[\"'`];?\s*?union\b\s*?distinct\b\s*?[^\s]
-[\"'`];?\s*?union\b\s*?select\b\s*?[^\s]
+##!> define quotes [\"'`]
+
+{{quotes}}\s*!\s*[\"'`\w]
```
Comment (theseion) [inline regex-assembly/942190.ra:8]: Use an assemble block with `{{quotes}}` as prefix so you don't have to repeat it on each of these lines:
```
##!> assemble
  {{quotes}}
  ##!=>
  ...
##!<
```

File: regex-assembly/942190.ra [hunk 2]
```diff
@@ -3,25 +3,27 @@
...
+\bconnection_id\s*?\(
+\bcurrent_user\s*?\(
+\bdatabase\s*?\(
 \bexec\s+master\.
 \bexecute\s+master\.
 \bfrom\W+information_schema\W
-\binto[\s+]+dumpfile\s*?[\"'`]
-\binto[\s+]+outfile\s*?[\"'`]
-\bschema\s*?\([^\)]*?
+\binto[\s+]+dumpfile\s*?{{quotes}}
```
Comment (theseion) [inline regex-assembly/942190.ra:23]: Same here, as above: use a block with `{{quotes}}` as suffix.

**Review by theseion — CHANGES_REQUESTED** [review]

**Comment by theseion** [comment]
I've taken over this PR and need someone else to do the review.

**Comment by dune73** [comment]
I'm not totally sure, but I get the feeling that `[^\)]*?` is missing from the following 3 original items:

* `\bconnection_id\s*?\([^\)]*?`
* `\bcurrent_user\s*?\([^\)]*?`
* `\bdatabase\s*?\([^\)]*?`

**Comment by theseion** [comment]
Sorry, the discussion was just between me and @53cur3M3. You're right, I should have added that to the PR.

**Comment by dune73** [comment]
No worries. That's why we do reviews after all. And now it's all documented.

**Comment by fzipi** [comment]
@53cur3M3 @theseion What's next here?

**Comment by theseion** [comment]
I just need a review :)

**Review by fzipi — APPROVED** [review]
LGTM!

**Review by theseion — APPROVED** [review]

**Pull request merged**